src/ngAnimate/animation.js

```diff
@@ -219,7 +235,32 @@ var $$AnimationProvider = ['$animateProvider', function($animateProvider) {
         // we need to sort each of the animations in order of parent to child
         // relationships. This ensures that the child classes are applied at the
         // right time.
-        $$rAFScheduler(sortAnimations(toBeSortedAnimations));
+        var finalAnimations = sortAnimations(toBeSortedAnimations);
+        for (var i = 0; i < finalAnimations.length; i++) {
```

Comment:
Is it worth it to have this loop where all it does it filter out the first (or only) animation and to not give it a prepareClass?